File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Ruby/Windows/X64/utils/runtime_name_handler.rb
# frozen_string_literal: true
require_relative './runtime_name'

class RuntimeNameHandler
  def self.get_name(runtime_name)
    if runtime_name == RuntimeName::CLR
      return 'clr'
    end
    if runtime_name == RuntimeName::GO
      return 'go'
    end
    if runtime_name == RuntimeName::JVM
      return 'jvm'
    end
    if runtime_name == RuntimeName::NETCORE
      return 'netcore'
    end
    if runtime_name == RuntimeName::PERL
      return 'perl'
    end
    if runtime_name == RuntimeName::PYTHON
      return 'python'
    end
    if runtime_name == RuntimeName::RUBY
      return 'ruby'
    end
    if runtime_name == RuntimeName::NODEJS
      return 'nodejs'
    end
    if runtime_name == RuntimeName::CPP
      return 'cpp'
    end
  end
end

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Nodejs/utils/RuntimeNameHandler.js
const RuntimeName = require('./RuntimeName');

class RuntimeNameHandler {
    static getName(runtimeName) {
        switch (runtimeName) {
            case RuntimeName.Clr:
                return "clr";
            case RuntimeName.Go:
                return "go";
            case RuntimeName.Jvm:
                return "jvm";
            case RuntimeName.Netcore:
                return "netcore";
            case RuntimeName.Perl:
                return "perl";
            case RuntimeName.Python:
                return "python";
            case RuntimeName.Ruby:
                return "ruby";
            case RuntimeName.Nodejs:
                return "nodejs";
            case RuntimeName.Cpp:
                return "cpp";
            default:
                throw new Error("Invalid runtime name.");
        }
    }
}

module.exports = RuntimeNameHandler;